src/reanimated2/mutables.ts

```diff
@@ -82,9 +82,9 @@ export function makeMutable<T>(initial: T): SharedValue<T> {
         })();
       }
     },
-    get value() {
+    get value(): T {
       if (!SHOULD_BE_USE_WEB) {
-        return executeOnUIRuntimeSync(uiValueGetter)(mutable);
+        return uiValueGetter(mutable) as T;
```

Comment:
![Screenshot 2023-12-21 at 18 48 17](https://github.com/software-mansion/react-native-reanimated/assets/40713406/b94e8127-5da5-4e63-ab71-38be46c5bc8d)

with the above change you don't need a cast here
```suggestion
        return uiValueGetter(mutable);
```